advanced_functionality/autogluon-sagemaker-pipeline/README.md

```diff
@@ -0,0 +1,110 @@
+
+A description of some of the artifacts is provided below:
+<br/><br/>
+Your codebuild execution instructions. This file contains the instructions needed to kick off an execution of the SageMaker Pipeline in the CICD system (via CodePipeline). You will see that this file has the fields definined for naming the Pipeline, ModelPackageGroup etc. You can customize them as required.
+
+```
+|-- codebuild-buildspec.yml
+```
+
+<br/><br/>
+Your pipeline artifacts, which includes a pipeline module defining the required `get_pipeline` method that returns an instance of a SageMaker pipeline, a preprocessing script that is used in feature engineering, and a model evaluation script to measure the Mean Squared Error of the model that's trained by the pipeline. This is the core business logic, and if you want to create your own folder, you can do so, and implement the `get_pipeline` interface as illustrated here.
+
+```
+|-- pipelines
+|   |-- abalone
+|   |   |-- evaluate.py
+|   |   |-- __init__.py
+|   |   |-- pipeline.py
+|   |   `-- preprocess.py
+
+```
+<br/><br/>
+Utility modules for getting pipeline definition jsons and running pipelines (you do not typically need to modify these):
+
+```
+|-- pipelines
+|   |-- get_pipeline_definition.py
+|   |-- __init__.py
+|   |-- run_pipeline.py
+|   |-- _utils.py
+|   `-- __version__.py
+```
+<br/><br/>
+Python package artifacts:
+```
+|-- setup.cfg
+|-- setup.py
+```
+<br/><br/>
+A stubbed testing module for testing your pipeline as you develop:
+```
+|-- tests
+|   `-- test_pipelines.py
+```
+<br/><br/>
+The `tox` testing framework configuration:
+```
+`-- tox.ini
+```
+
+## Dataset for the Example Abalone Pipeline
+
+The dataset used is the [UCI Machine Learning Abalone Dataset](https://archive.ics.uci.edu/ml/datasets/abalone) [1]. The aim for this task is to determine the age of an abalone (a kind of shellfish) from its physical measurements. At the core, it's a regression problem. 
+    
+The dataset contains several features - length (longest shell measurement), diameter (diameter perpendicular to length), height (height with meat in the shell), whole_weight (weight of whole abalone), shucked_weight (weight of meat), viscera_weight (gut weight after bleeding), shell_weight (weight after being dried), sex ('M', 'F', 'I' where 'I' is Infant), as well as rings (integer).
+
+The number of rings turns out to be a good approximation for age (age is rings + 1.5). However, to obtain this number requires cutting the shell through the cone, staining the section, and counting the number of rings through a microscope -- a time-consuming task. However, the other physical measurements are easier to determine. We use the dataset to build a predictive model of the variable rings through these other physical measurements.
+
+We'll upload the data to a bucket we own. But first we gather some constants we can use later throughout the notebook.
```

Comment:
Delete this line as it's not relevant in the context of the readme?

Reply:
done!